UpgradedVehicles/VehicleUpgrader.cs

```diff
@@ -340,9 +354,16 @@ internal void UpgradeVehicle(TechType upgradeModule)
             }
 
             int nextDepthIndex = CalculateDepthModuleIndex();
-
+            TechType[] armorPlatingModules =
+            {
+                TechType.VehicleArmorPlating,
+                HullArmorMk2.HullArmorMk2TechType,
+                HullArmorMk3.HullArmorMk3TechType,
+                HullArmorMk4.HullArmorMk4TechType
+            };
```

Comment:
This TechType array is going to be re-used a lot and won't ever change.  
Rather than create it in-line, instead define it as a `static readonly` member same as `CommonUpgradeModules`.  

Also, explicitly using `HashSet<TechType>` or `List<TechType>` will like you use `Contains` or `Find` natively, avoiding the need to include `Linq`.

Make sure to account for the fact that these TechType values are defined at runtime, similar to what you've done with `SetNewModule`.